.travis.yml

```diff
@@ -17,12 +17,10 @@ jdk:
   - oraclejdk8
 
 install:
-  - mvn clean install -DskipTests -Dassembly.skipAssembly=true > /dev/null
+  - bash .travis_install.sh
 
 script:
-  - cd $PINOT_MODULE
-  - cp ../.travis_cmd.sh .
-  - bash .travis_cmd.sh
+  - bash .travis_test.sh
 
 after_success:
   - bash <(curl -s https://codecov.io/bash)
```

Comment:
Beyond a point, this will degenerate to academic discussion with no
actionable value....

Security-wise, the biggest thing we should worry about, violation of github
security to push commits to our code. This has happened on github in the
past.

By github compromise, I was talking of our scripts being compromised. Even
git database can be compromised if we don't use signed verifications.
Jackie can do an year long project on this ;)
Their servers doing DDoS because the script on their server is
compromised...I would say acceptable risk for the benefit of staying up to
date.



On Thu, Jun 2, 2016 at 1:08 PM, Jean-François Im <notifications@github.com>
wrote:

> In .travis.yml
> <https://github.com/linkedin/pinot/pull/140#discussion_r65609372>:
>
> >
> >  script:
> > -  - cd $PINOT_MODULE
> > -  - cp ../.travis_cmd.sh .
> > -  - bash .travis_cmd.sh
> > +  - bash .travis_test.sh
> >
> >  after_success:
> >    - bash <(curl -s https://codecov.io/bash)
>
> There is a difference. If you git clone 1234567 by definition either hash
> 1234567 exists and has the content that you expect, unless someone has
> created malicious content that happens to have the exact same SHA-1 hash.
> This is unlikely, but not impossible (see
> https://www.schneier.com/blog/archives/2012/10/when_will_we_se.html).
>
> Pulling it from an external server adds the risk that the server is
> compromised somehow and serves a completely different version; checking out
> a known revision eliminates that risk as we can vet the contents of the
> script ahead of time and, barring a hash collision attack, that someone
> inserts code in that script.
>
> I'd replace the bash/curl call by:
>
> git clone https://github.com/codecov/codecov-bash.git
> git checkout d3157145db519e2603b870e9596b9086a7983ac1 && so codecov/codecov
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/linkedin/pinot/pull/140/files/0b806a490eba7a6250c08dab0ce795029c2f111e#r65609372>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABAmRoEnsA1JhbPbBG_maBnpi2UHJMzwks5qHzhPgaJpZM4IsJ6Y>
> .
>